stdlib/asm/crypto/stark/ood_frames.masm

```diff
@@ -16,100 +16,103 @@ use.std::crypto::stark::random_coin
 #! 4. alpha is the randomness used in order to build the DEEP polynomial.
 export.load_and_horner_eval_ood_frames
-    # I) Load the random challenge used in computing the DEEP polynomial.
-    #    We use this challenge to compute the constant terms needed in the computation of the DEEP queries.
-    #    Although this challenge is generated only after all the OOD evaluations are received by the verifier,
-    #    we use non-determinism to generate it before doing so. This is done so that we can hash, memory store
-    #    and Horner evaluate in parallel.
+    # --- Load the random challenge used in computing the DEEP polynomial -----
+
+    # We use this challenge to compute the constant terms needed in the computation of the DEEP
+    # queries. Although this challenge is generated only after all the OOD evaluations are received
+    # by the verifier, we use non-determinism to generate it before doing so. This is done so that
+    # we can hash, memory store and Horner evaluate in parallel.
     
-    ## 1) Load the random challenge non-deterministically
+    ## Load the random challenge non-deterministically
     adv_push.2
-    # => [alpha_1, alpha_0, ...]
+    # => [alpha_1, alpha_0]
 
-    ## 2) Save the random challenge
+    # Save the random challenge
     dup.1 dup.1
     exec.constants::deep_rand_alpha_nd_ptr mem_storew
-    # => [Y, ...]
+    # => [Y]
 
-    # II) Compute Q^z(alpha)
+    # --- Compute Q^z(alpha) --------------------------------------------------
 
     ## 1) Set up the stack for `horner_eval_ext` to compute Q^z(alpha)
 
-    ### a) Set up the initial accumulator and the pointers to alpha and a pointer to some memory region
-    ###    to which we save the OOD.
+    ### a) Set up the initial accumulator and the pointers to alpha and a pointer to some memory
+    ###    region to which we save the OOD.
     push.0.0
     exec.constants::deep_rand_alpha_nd_ptr
     exec.constants::ood_evaluations_ptr
-    # => [ood_evaluations_ptr, deep_rand_alpha_ptr, 0, 0, Y, ...]
-    # => [U, Y, ...]
+    # => [ood_evaluations_ptr, deep_rand_alpha_ptr, 0, 0, Y]
+    # => [U, Y]
 
     ## 2) Process the fully aligned OOD `current` evaluations at z of the execution trace
     ##    and constraints polynomials evaluations. 
-    ##    Since there are (80 + 8 + 8) * 2 = 24 * 8 base field elements, there are 24 fully double-word aligned batches.
-    ## Note: the first word is the capacity, where its first element is initialized with the number of elements to hash MODULO 8.
+    ##    Since there are (80 + 8 + 8) * 2 = 24 * 8 base field elements, there are 24 fully
+    ##    double-word aligned batches.
+    ## Note: the first word is the capacity, where its first element is initialized with the number
+    ## of elements to hash MODULO 8.
 
     ### a) Set up the hasher state
-    padw
-    padw     
-    # => [ZERO, 0, 0, 0, 0, U, Y, ...]
+    padw padw     
+    # => [ZERO, 0, 0, 0, 0, U, Y]
```

Comment:
Seems like this code makes assumptions about trace width and constraint degrees, and so, is probably not generic enough. We have at least two potential solutions

- Have another dynamic function initialize the hasher state.
- Pass in trace width and constraint degree to figure out how to initialize the capacity in this procedure.
- Maybe something else?

Reply:
This is actually just an out-of-place comment. There is nothing as far as I can see that is specific here to the VM verifier.

Reply:
We have discussed this a while but, at the moment, we just pad the traces to the next multiple of 8 with 0 filled columns. Based on benchmarks, by myself and Phil, this doesn't incur any noticeable performance overhead.

Reply:
Indeed, forgot memory as well, thanks for bringing it up